src/components/common/Header/Header.jsx

```diff
@@ -1,9 +1,41 @@
-import hamburger from 'assets/hamburger_icon.svg';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import routesPaths from 'routes/routesPaths';
+import hamburger from 'assets/hamburger.svg';
+import close from 'assets/close.svg';
 
-const Header = () => (
-  <header className="header">
-    <img src={hamburger} alt="menu icon" className="header__icon--hamburger" />
-  </header>
-);
+const Header = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
```

Comment:
```suggestion
    setIsOpen(current => !current);
```

Reply:
I don't understand why is it better to do it this way.

Reply:
setting the state is asynchronous, since you rely on the previous state it's better to use this. that way you always have the right value when you do the call